pkgs/by-name/op/openmpi/package.nix

```diff
@@ -147,122 +146,42 @@ stdenv.mkDerivation (finalAttrs: {
-      };
-    in
-    ''
-      find $out/lib/ -name "*.la" -exec rm -f \{} \;
-
-      # The main wrapper that all the rest of the commonly used binaries are
-      # symlinked to
-      moveToOutput "bin/opal_wrapper" "''${!outputDev}"
-      # All of the following files are symlinks to opal_wrapper
-      ${lib.pipe fileNamesToIterate [
-        (lib.mapCartesianProduct (
-          { p, s }:
-          ''
-            echo "handling ${p}${s}"
-            moveToOutput "bin/${p}${s}" "''${!outputDev}"
-            moveToOutput "share/openmpi/${p}${s}-wrapper-data.txt" "''${!outputDev}"
-          ''
-        ))
-        (lib.concatStringsSep "\n")
-      ]}
-      # default compilers should be indentical to the
-      # compilers at build time
-      ${lib.pipe wrapperDataFileNames [
-        (lib.mapCartesianProduct (
-          { part1, part2 }:
-          # From some reason the Darwin build doesn't include some of these
-          # wrapperDataSubstitutions strings and even some of the files. Hence
-          # we currently don't perform these substitutions on other platforms,
-          # until a Darwin user will care enough about this cross platform
-          # related substitution.
-          lib.optionalString stdenv.hostPlatform.isLinux ''
-            substituteInPlace "''${!outputDev}/share/openmpi/${part1}${part2}-wrapper-data.txt" \
-              --replace-fail \
-                compiler=${lib.elemAt wrapperDataSubstitutions.${part2} 0} \
-                compiler=${lib.elemAt wrapperDataSubstitutions.${part2} 1}
-          ''
-        ))
-        (lib.concatStringsSep "\n")
-      ]}
-      # A symlink to $\{lib.getDev pmix}/bin/pmixcc upstreeam puts here as well
-      # from some reason.
-      moveToOutput "bin/pcc" "''${!outputDev}"
-
-      # Handle informative binaries about the compilation
-      for i in {prte,ompi,oshmem}_info; do
-        moveToOutput "bin/$i" "''${!outputDev}"
-      done
-    '';
+  postInstall = ''
+    find $out/lib/ -name "*.la" -exec rm -f \{} \;
+
+    # The main wrapper that all the rest of the commonly used binaries are
+    # symlinked to
+    moveToOutput "bin/opal_wrapper" "''${!outputDev}"
+    # All of the following files are symlinks to opal_wrapper
+    find $out/bin -type l -lname "opal_wrapper" -exec mv {} "''${!outputDev}/bin" \;
+    # default compilers should be indentical to the
+    # compilers at build time
+    moveToOutput "share/openmpi/*-wrapper-data.txt" "''${!outputDev}"
+    find "''${!outputDev}/share/openmpi" -type f -name "*-wrapper-data.txt" -print0 \
+      | xargs -0 sed -i \
+      -e "s,compiler=$CC,compiler=${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}$CC," \
+      -e "s,compiler=$CXX,compiler=${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}$CXX," \
+      -e "s,compiler=$FC,compiler=${targetPackages.gfortran}/bin/${targetPackages.gfortran.targetPrefix}$FC,"
```

Comment:
I very much dislike these `sed` commands - you have no indication if they actually do something or not, and their syntax is too complicated for our task, since we don't need any regular expressions benefits. I specifically switched from this to the seemingly complicated Nix expression based substitution in [`36494fe` (#327438)](https://github.com/NixOS/nixpkgs/pull/327438/commits/36494fe30f07f33c3d456e0e0c546ca13a6c6306) .